sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/ConstantColumnVector.java

```diff
@@ -70,10 +70,15 @@ public ConstantColumnVector(int numRows, DataType type) {
 
   @Override
   public void close() {
+    stringData = null;
```

Comment:
This change is not related to the NPE fix, but it's good to have. We should release the memory for `UTF8String` as well.